pkg/controller/common/association/association.go

```diff
@@ -58,7 +58,7 @@ func writeAuthSecretToConfigHash(client k8s.Client, assoc commonv1.Association,
 
 func writeCASecretToConfigHash(client k8s.Client, assoc commonv1.Association, configHash hash.Hash) error {
 	assocConf := assoc.AssociationConf()
-	if !assocConf.CAIsConfigured() {
+	if !assocConf.GetCACertProvided() {
```

Comment:
I am still confused by this. If all we care about is the CA certificate why do we need these two different methods? Why do we copy the public certificates at all into the associations namespace when they do not contain a CA? What I mean is could we not simplify this further?